tests/lib/config/config-validator.js

```diff
@@ -102,116 +102,249 @@ describe("Validator", () => {
+                assert.doesNotThrow(fn);
+            });
 
-            assert.doesNotThrow(fn);
         });
 
-        it("should not allow options for rules with no options", () => {
-            eslint.defineRule("mock-no-options-rule", mockNoOptionsRule);
+        describe("plugins", () => {
+            it("should not throw with an empty array", () => {
+                const fn = validator.validate.bind(null, { plugins: [] });
 
-            const fn = validator.validate.bind(null, { rules: { "mock-no-options-rule": [2, "extra"] } }, "tests");
+                assert.doesNotThrow(fn);
+            });
+
+            it("should throw with a string", () => {
+                const fn = validator.validate.bind(null, { plugins: "react" });
 
-            assert.throws(fn, "tests:\n\tConfiguration for rule \"mock-no-options-rule\" is invalid:\n\tValue \"extra\" has more items than allowed.\n");
+                assert.throws(fn, "Property \"plugins\" is the wrong type (expected array but got `\"react\"`).");
+            });
         });
 
-        it("should throw with an array environment", () => {
-            const fn = validator.validate.bind(null, { env: [] });
+        describe("settings", () => {
+            it("should not throw with an empty object", () => {
+                const fn = validator.validate.bind(null, { settings: {} });
 
-            assert.throws(fn, "Environment must not be an array");
-        });
+                assert.doesNotThrow(fn);
+            });
 
-        it("should throw with a primitive environment", () => {
-            const fn = validator.validate.bind(null, { env: 1 });
+            it("should throw with an array", () => {
+                const fn = validator.validate.bind(null, { settings: ["foo"] });
 
-            assert.throws(fn, "Environment must be an object");
+                assert.throws(fn, "Property \"settings\" is the wrong type (expected object but got `[\"foo\"]`).");
+            });
         });
 
-        it("should catch invalid environments", () => {
-            const fn = validator.validate.bind(null, { env: { browser: true, invalid: true } });
+        describe("extends", () => {
+            it("should not throw with an empty array", () => {
+                const fn = validator.validate.bind(null, { extends: [] });
+
+                assert.doesNotThrow(fn);
+            });
+
+            it("should not throw with a string", () => {
+                const fn = validator.validate.bind(null, { extends: "react" });
+
+                assert.doesNotThrow(fn);
+            });
+
+            it("should throw with an object", () => {
+                const fn = validator.validate.bind(null, { extends: {} });
 
-            assert.throws(fn, "Environment key \"invalid\" is unknown\n");
+                assert.throws(fn, "Property \"extends\" is the wrong type (expected string,array but got `{}`).");
```

Comment:
Wondering if we can improve the expected type display by transliterating commas to forward slashes?